systers_portal/meetup/mixins.py

```diff
@@ -0,0 +1,24 @@
+from django.core.exceptions import ImproperlyConfigured
+
+
+class MeetupLocationMixin(object):
+    """Mixin to add information about MeetupLocation to context, as per the slug"""
+    meetup_location = None
```

Comment:
So to get a MeetupLocation you only need the slug, right? In this case you can set not the object itself, but the meetup location as a slug. And then make a query right inside the MeetupLocationMixin.

There is a tradeoff, because you need to think how to do it in a way that you query the MeetupLocation only once. Please think about it and if you come up with use cases where in a view you will need the MeetupLocation object again, write it here. If not, it's good to go.